### PR TITLE
consensus: fix height advances in test state

### PR DESCRIPTION
**Pull request by williambanfield**
The problem with the `TestStateFullRound1` is that the state that we are observeing, `cs`, can advance to the next height before we query its data. Specifically, on line `388`, when we called `validatePrevote`, the `cs` State had already advanced to height 2, so querying that State for the votes of height 1 either yielded nil or an erroneous value. This change adds a `ensurePrevoteMatch` function that checks that the prevote occurred and checks that it is for the expected block at the same time. If this change looks reasonable I can just apply the same fix to all of the places where we perform `ensurePrevote` followed by `validatePrevote` to use this function instead.

### CODE REVIEW

File: internal/consensus/common_test.go
Comment (tychoish) [inline internal/consensus/common_test.go:696]: should we also pass a context to this? 

(I also get edgy about time.After, because they don't cleanup the goroutine on the other side of the channel until the timeout fires, which might trigger the leak detector)

**Review by tychoish — APPROVED** [review]

File: internal/consensus/state_test.go
Comment (williambanfield) [inline internal/consensus/state_test.go:356]: todo: revert

**Comment by tychoish** [comment]
is this change applicable to `TestStateFullRoundNil`?

**Comment by williambanfield** [comment]
> is this change applicable to `TestStateFullRoundNil`?

Yes.

**Review by creachadair — APPROVED** [review]
_Pace_ the aforementioned, LGTM.

File: internal/consensus/common_test.go
Comment (williambanfield) [reply internal/consensus/common_test.go:685]: This is the current way the other `ensure*` methods work. Selfishly, I'd prefer to wait until after the PBTS merge happens in the next few days so that I don't have to do another round of conflict fixes and update all these `ensure*` functions in one go.

Comment (williambanfield) [reply internal/consensus/common_test.go:685]: https://github.com/tendermint/tendermint/issues/7649

**Pull request merged**
